gematria/datasets/process_and_filter_bbs.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+
+#include "gematria/llvm/disassembler.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Error.h"
+
+using namespace llvm;
+
+static cl::opt<std::string> InputFile(
+    "input-file",
+    cl::desc("Path to the input CSV file containing hex basic blocks"),
+    cl::init(""));
+
+static cl::opt<std::string> OutputFile(
+    "output-file",
+    cl::desc(
+        "Path to the output CSV file with processed/filtered basic blocks"),
+    cl::init(""));
+
+Expected<std::string> ProcessBasicBlock(
```

Comment:
Very nitpicky: `processBasicBlock` to be in line with the LLVM naming conventions.